File: src/main/java/com/lcaohoanq/nocket/domain/chat/Attachment.java
package com.lcaohoanq.nocket.domain.chat;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.lcaohoanq.nocket.base.entity.BaseEntity;
import com.lcaohoanq.nocket.metadata.MediaMeta;
import jakarta.persistence.AttributeOverride;
import jakarta.persistence.AttributeOverrides;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "attachments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@AttributeOverrides({
    @AttributeOverride(name = "createdAt", column = @Column(name = "attachment_created_at")),
    @AttributeOverride(name = "updatedAt", column = @Column(name = "attachment_updated_at"))
})
public class Attachment extends BaseEntity {
    
    @Embedded
    private MediaMeta mediaMeta;
    
}

File: src/main/java/com/lcaohoanq/nocket/domain/chat/Attachment.kt
package com.lcaohoanq.nocket.domain.chat

import com.lcaohoanq.nocket.base.entity.BaseEntity
import com.lcaohoanq.nocket.metadata.MediaMeta
import jakarta.persistence.*
import org.hibernate.envers.Audited

@Audited
@Entity
@Table(name = "attachments")
@AttributeOverrides(
    AttributeOverride(
        name = "createdAt",
        column = Column(name = "attachment_created_at")
    ), AttributeOverride(name = "updatedAt", column = Column(name = "attachment_updated_at"))
)
class Attachment : BaseEntity() {
    @Embedded
    private val mediaMeta: MediaMeta? = null
}
